luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs

```diff
@@ -208,6 +208,85 @@ impl Component for WysiwygEditor<'_> {
                 } => {
                     let image_width_per_height_ratio = graphic_wh.width / graphic_wh.height;
                     builder
+                        .add_button("Send To Back", || {
+                            let Some(last_graphic_index) =
+                                screen_graphics.last().map(|(index, _)| *index)
+                            else {
+                                return;
+                            };
+
+                            SEQUENCE_ATOM.mutate(move |sequence| {
+                                sequence.update_cut(
+                                    cut_id,
+                                    CutUpdateAction::ChangeGraphicOrder {
+                                        graphic_index,
+                                        after_graphic_index: Some(last_graphic_index),
+                                    },
+                                )
+                            });
+                        })
+                        .add_button("Send Backward", || {
+                            let Some(next_or_last_graphic_index) = ({
+                                let Some(position) = screen_graphics
+                                    .iter()
+                                    .position(|(index, _)| *index == graphic_index)
+                                else {
+                                    return;
+                                };
+                                let next_position = (position + 1).min(screen_graphics.len() - 1);
+                                screen_graphics.get(next_position).map(|(index, _)| *index)
```

Comment:
Personally looks so ugly :( using `map` fn must be better

Reply:
```rust
screen_graphics
    .iter()
    .position(|(index, _)| *index == graphic_index)
    .map(|position| {
        let next_position =
            (position + 1).min(screen_graphics.len() - 1);
        screen_graphics.get(next_position).map(|(index, _)| *index)
    })
```

I've been trying to reduce depth recently.
But, yes, it looks much better.

Reply:
Fixed with [291a7eb](https://github.com/NamseEnt/namseent/pull/681/commits/291a7eb804658b3235e473217a04dc5dffab2644)